services/smartcards/testCardReader.py

```diff
@@ -1,36 +1,100 @@
-
-import json, hashlib, secrets, time, sys
+import re
+import json
+import hashlib
+import secrets
+import time
+import sys
 from smartcards.core import CardInterface
+import smartcard
+
+REFRESH_INTERVAL = 0.2
+
+
+def wait_for_reader():
+    if not CardInterface.is_reader_connected():
+        print("Insert card reader")
+    while True:
+        time.sleep(REFRESH_INTERVAL)
+        if CardInterface.is_reader_connected():
+            break
+    print("Card reader connected")
+
+
+def wait_for_card():
+    if not CardInterface.card:
+        print("Insert card")
+    while True:
+        time.sleep(REFRESH_INTERVAL)
+        if CardInterface.card:
+            break
```

Comment:
Some cards are write-protected with a bit that we set and look for, and in that case the tests will fail. This minimal change just exits if the card is write-protected:

```suggestion
            if not CardInterface.card.write_enabled:
                print("card is write-protected, please use a different card.")
                sys.exit(1)
            break
```

Reply:
added 8a0b05c7